cmd/ci-operator-configresolver/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/interrupts"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+)
+
+type options struct {
+	configs    filenameToConfig
+	configPath string
+}
+
+type filenameToConfig map[string]api.ReleaseBuildConfiguration
+
+func resolveConfig(o *options) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusNotImplemented)
+			w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+		org := r.URL.Query().Get("org")
+		if org == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("org query missing or incorrect"))
+			return
+		}
+		repo := r.URL.Query().Get("repo")
+		if repo == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("repo query missing or incorrect"))
+			return
+		}
+		branch := r.URL.Query().Get("branch")
+		if branch == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("branch query missing or incorrect"))
+			return
+		}
+		variant := r.URL.Query().Get("variant")
+		info := config.Info{
+			Org:     org,
+			Repo:    repo,
+			Branch:  branch,
+			Variant: variant,
+		}
+		info.Filename = filepath.Join(o.configPath, branch, info.Basename())
+
+		config, ok := o.configs[info.Basename()]
+		if !ok {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("config does not exist"))
+			return
+		}
+		jsonConfig, err := json.Marshal(config)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "failed to marshal config to JSON: %v", err)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Write(jsonConfig)
+	}
+}
+
+// loadFilenameToConfig generates a new filenameToConfig map.
+func loadFilenameToConfig(configPath string) (filenameToConfig, error) {
+	log.Debug("Reloading configs")
+	configs := filenameToConfig{}
+	err := filepath.Walk(configPath, func(path string, info os.FileInfo, err error) error {
+		ext := filepath.Ext(path)
+		if info != nil && !info.IsDir() && (ext == ".yml" || ext == ".yaml") {
+			data, err := ioutil.ReadFile(path)
+			if err != nil {
+				return fmt.Errorf("failed to read ci-operator config (%v)", err)
+			}
+
+			var configSpec *api.ReleaseBuildConfiguration
+			if err := yaml.Unmarshal(data, &configSpec); err != nil {
+				return fmt.Errorf("failed to load ci-operator config (%v)", err)
+			}
+
+			if err := configSpec.ValidateAtRuntime(); err != nil {
+				return fmt.Errorf("invalid ci-operator config: %v", err)
+			}
+			log.Debugf("Adding %s to filenameToConfig", filepath.Base(path))
+			configs[filepath.Base(path)] = *configSpec
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	log.Info("Configs reloaded")
+	return configs, nil
+}
+
+func main() {
+	opts := options{}
```

Comment:
check [how](https://github.com/openshift/ci-tools/blob/master/cmd/pj-rehearse/main.go#L44-L79) we are used to parsing the flags.